presto-main/src/main/java/com/facebook/presto/server/QueryExecutionResource.java

```diff
@@ -121,7 +121,7 @@ public Response getTaskInfo(@PathParam("queryId") String queryId)
 
                 tasks.add(new Task(taskStatus.getTaskId().toString(),
                         taskStatus.getState().toString(),
-                        taskStatus.getSelf().getHost(),
+                        taskStatus.getSelf().getHost() + ":" + taskStatus.getSelf().getPort(),
```

Comment:
This field in `Task` should become a `HostAddress` which can be created using `HostAddress.fromUri`